### ISSUE TITLE
Issue installing using devtools - SDMTools not available

### ISSUE THREAD

**Issue by Jakob37** [IFIproteomics/LFQbench#8]
Dear developers,

I tried install LFQbench using the devtools as outlined in the README, but it seems like one of the dependent packages is no longer maintained.

```{r}
> devtools::install_github("IFIproteomics/LFQbench")
Downloading GitHub repo IFIproteomics/LFQbench@master
Skipping 1 packages not available: SDMTools
Installing 4 packages: hash, SDMTools, classInt, pander
Installing packages into ‘/home/user/R/x86_64-pc-linux-gnu-library/3.6’
(as ‘lib’ is unspecified)
Error: Failed to install 'LFQbench' from GitHub:
  (converted from warning) package ‘SDMTools’ is not available (for R version 3.6.3)
```

When checking the SDMTools package I see that it has been removed from CRAN since January 2020:

https://cran.r-project.org/web/packages/SDMTools/index.html

I succeeded in installing LFQbench after downloading an archived version of the package and installing it locally. So it works for me for now, just wanted to let you know.

**Comment by cgonzalez87-uk** [comment]
Hi,

I had the same issue with the SDMTools package, so I installed the last archived version after installing RTools. It worked well, but now I am getting this error:


* installing *source* package 'LFQbench' ...
** using staged installation
** R
** inst
** byte-compile and prepare package for lazy loading
Error: (converted from warning) package 'hash' was built under R version 3.6.3
Execution halted
ERROR: lazy loading failed for package 'LFQbench'
* removing 'C:/Users/cgonzalez//Documents/R/win-library/3.6/LFQbench'
installation of package �C:/Users/CARMEN~1.GON/AppData/Local/Temp/Rtmpe2cZ3z/file63186ccc4519/LFQbench_1.0.0.tar.gz� had non-zero exit status


Any ideas on how to fix this issue?

Thanks very much in advance!

**Comment by MatteoLacki** [comment]
Hi,

I'm not a developer of this package, but it sort of looks (to me) like a
clash of versions of R and package 'hash'.
Maybe try installing R >= 4 and then reinstalling the package?
I get a lot of errors with R whenever the wise men of R update anything.
What version of R do you have now installed?
Is it a clean install?

Best wishes,

*Mateusz Krzysztof Łącki*

*German tel. +49 159 01681376*
Polish tel. +48 579 647 311
Skype:   mathewin
GitHub: MatteoLacki <https://github.com/MatteoLacki>

**Comment by cgonzalez87-uk** [comment]
Hi Mateusz,

Thank you so much for your reply. I installed the last version for R (v4.0.1) and also installed RTools40 (https://cran.r-project.org/bin/windows/Rtools/). After that, I installed the last archived version of SDMTools (SDMTools_1.1-221.2.tar) and finally the LFQBench package...and everything went very well this time!

Thank you so much for your help.

Best wishes,

Carmen

**Comment by MatteoLacki** [comment]
Glad to help. R is a terribly maintained *language* and staying up to date
is the only way to get things to work.
If you cannot afford it, there is this package called  PACKRAT, which in
principle advertises that it can work with older versions of packages.
But I have little experience with it, so I don't know if I can recommend it.

Best wishes!

*Mateusz Krzysztof Łącki*

*German tel. +49 159 01681376*
Polish tel. +48 579 647 311
Skype:   mathewin
GitHub: MatteoLacki <https://github.com/MatteoLacki>

**Comment by cgonzalez87-uk** [comment]
Thanks very much! Good to know just in case :)

Best wishes,

Carmen

**Comment by mesontau** [comment]
Hi. I am sorry about this issue with the SDMTools package. It is used for plotting a couple of figures. I will see if I can substitute its functionality. I agree with Matteo: one of the worst things of R is about its maintenance.

**Comment by MatteoLacki** [comment]
Well, it works after upgrading :)
I have a lot of problems myself trying to keep R packages minimal and
suckless..
One of bigger things to eradicate would be ggplot itself, but, well, trying
to get other plots to look that nice at such low costs is difficult.

*Mateusz Krzysztof Łącki*

*German tel. +49 159 01681376*
Polish tel. +48 579 647 311
Skype:   mathewin
GitHub: MatteoLacki <https://github.com/MatteoLacki>

**Comment by mesontau** [comment]
Hi Matteo,
I upgraded the requirement of R version to 4.0.1. I hope this solves the issue to everyone.

**Comment by hcji** [comment]
> Hi Matteo,
> I upgraded the requirement of R version to 4.0.1. I hope this solves the issue to everyone.

However, I found the latest version of SDMTools (1.1-221) requires R < 4.0.0. I tried with R 3.6.3 and installed SDMTools with source, then removed the requirement of R version of LFQbench. It worked.
Is there any better way?

**Comment by mesontau** [comment]
Hi HC.Ji and Matteo,

yes, there is a better way. I removed the dependency of SDMTools package. It was only used in order to calculate the area under the curve in a binary system. Since SDMTools is no longer maintained and its license is GPL v3, it is allowed to use all its code freely. I copied the function for estimating the AUC, acknowledging to the authors of SDMTools. I will send a Pull Request today.
